src/main/java/cinema/dao/impl/CinemaHallDaoImpl.java

```diff
@@ -57,4 +58,18 @@ public CinemaHall add(CinemaHall cinemaHall) {
                     "Can't retrieve all cinema halls from the database", e);
         }
     }
+
+    @Override
+    public CinemaHall getById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            Query<CinemaHall> query = session.createQuery(
+                    "from CinemaHall where id = :id", CinemaHall.class);
+            query.setParameter("id", id);
+            LOGGER.info("Cinema hall with id " + id + " has been found in the database");
+            return query.uniqueResult();
```

Comment:
session.get
```suggestion
           return session.get(CinemaHall.class, id);
```
try this one here and everywhere(each dao) where you are getting entity by id